lib/vocabularies/validation/enum.ts

```diff
@@ -20,7 +21,11 @@ const def: CodeKeywordDefinition = {
     const {gen, data, $data, schema, schemaCode, it} = cxt
     if (!$data && schema.length === 0) throw new Error("enum must have non-empty array")
     const useLoop = schema.length >= it.opts.loopEnum
-    const eql = useFunc(gen, equal)
+    const needsEql =
```

Comment:
I've made a small correction, but in general this is a brittle approach - if the condition of usage changes, these conditions would have to be updated too, and there may be edge cases that would not be caught by the tests.

A much better approach would be to simply call useFunc in the place where eql is actually used.

Reply:
so probably just make a memoized getEql function and replace eql with getEql() in two places.